packages/video_player/video_player/ios/Classes/FLTVideoPlayerPlugin.m

```diff
@@ -331,25 +331,48 @@ - (void)updatePlayingState {
 
 - (void)setupEventSinkIfReadyToPlay {
   if (_eventSink && !_isInitialized) {
-    BOOL hasVideoTracks =
-        [[self.player.currentItem.asset tracksWithMediaType:AVMediaTypeVideo] count] != 0;
-    CGSize size = [self.player currentItem].presentationSize;
+    AVPlayerItem *currentItem = self.player.currentItem;
+    CGSize size = currentItem.presentationSize;
     CGFloat width = size.width;
     CGFloat height = size.height;
 
+    // Wait until tracks are loaded to check duration or if there are any videos.
+    AVAsset *asset = currentItem.asset;
+    if ([asset statusOfValueForKey:@"tracks" error:nil] != AVKeyValueStatusLoaded) {
+      void (^trackCompletionHandler)(void) = ^{
+        if ([asset statusOfValueForKey:@"tracks" error:nil] != AVKeyValueStatusLoaded) {
+          // Cancelled, or something failed.
+          return;
+        }
+        // This completion block will run on an unknown AVFoundation completion
+        // queue thread. Hop back to the main thread to set up event sink.
+        if (!NSThread.isMainThread) {
+          [self performSelector:_cmd onThread:NSThread.mainThread withObject:self waitUntilDone:NO];
+        } else {
+          [self setupEventSinkIfReadyToPlay];
```

Comment:
Is this never called?

Reply:
Maybe remove the `if` check? From [the API doc](https://developer.apple.com/documentation/avfoundation/avasynchronouskeyvalueloading/1387321-loadvaluesasynchronouslyforkeys): 

> The system calls the closure on a background queue, so you should dispatch control back to the main queue before performing any user interface-related operations.

Reply:
Seems safe, done.